File: eskybe/src/main/java/com/esky/service/ClassService.java
package com.esky.service;

import com.esky.model.entities.Class;
import com.esky.model.pojo.ClassRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface ClassService {

    //Save Class:
    public Class saveClass(ClassRequest aClass);

    //Update Class:
    public Class updateClass(ClassRequest aClass);

    //Delete Class:
    public void deleteClass(ClassRequest aClass);

    //Delete Class by ID:
    public void deleteClassById(Long id);

    //Get Class by ID:
    public Class getClassById(Long id);

    //Get all Classes by Filter:
    public Page<Class> findAllClassByFilter(PageRequest pageRequest, String filterValue);

    //Get all Classes by Creator:
    public Page<Class> findAllClassByCreator(PageRequest pageRequest, String filterValue, Long idCreator);

    //Get all Classes by Student:
    public Page<Class> findByStudentAndFilter(PageRequest pageRequest, String filterValue, Long idCreator);

    //Get my Classes by Creator:
    public List<ClassRequest> findMyClassByCreator(Long idCreator);

    //Find all Classes:
    public Page<Class> findAllClass(PageRequest pageRequest);

    //Get Courses Count By Creator:
    public long getCoursesCountByCreator(Long id);

    //Get Courses Count By Student:
    public long getCoursesCountByStudent(Long id);

    //Get Classes Count
    public long getClassesCount();

}

File: webui/src/main/webui/src/app/shared/services/http/class.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpParams} from '@angular/common/http';
import {HttpHeaders} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Class} from "../../entities/class";
import {api, environment} from "../../../../environments/environment";
import {User} from "../../entities/user";
import {Pageable} from "../../entities/pageable";

const httpOptions = {
    headers: new HttpHeaders({
        'Content-Type': 'application/json',
        'Authorization': 'my-auth-token'
    })
};

@Injectable()
export class ClassService {

    constructor(private http: HttpClient) {
    }

    getCountClasses() {
        return this.http.get<number>(environment.endpoint + api.class + api.count);
    }

    getAllClasses(pageable: Pageable) {
        const params = new HttpParams()
            .set('page', pageable.page)
            .set('size', pageable.size)
            .set('direction', pageable.direction)
            .set('sort', pageable.sort);
        return this.http.get<Class[]>(environment.endpoint + api.class + api.all, {params});
    }

    getClassesByFilter(pageable: Pageable, filterValue: string): any {
        const params = new HttpParams()
            .set('page', pageable.page)
            .set('size', pageable.size)
            .set('direction', pageable.direction)
            .set('sort', pageable.sort)
            .set('filterValue', filterValue);
        return this.http.get<Class[]>(environment.endpoint + api.class + api.all + api.filter, {params});
    }

    getClassesByCreator(pageable: Pageable, filterValue: string, idCreator: number): any {
        const params = new HttpParams()
            .set('page', pageable.page)
            .set('size', pageable.size)
            .set('direction', pageable.direction)
            .set('sort', pageable.sort)
            .set('filterValue', filterValue);
        return this.http.get<Class[]>(environment.endpoint + api.class + api.all + api.creator + idCreator, {params});
    }

    getMyClassesByCreator(idCreator: number): any {
        return this.http.get<Class[]>(environment.endpoint + api.class + api.mine + api.creator + idCreator);
    }

    getClassesByStudent(pageable: Pageable, filterValue: string, idStudent: number): any {
        const params = new HttpParams()
            .set('page', pageable.page)
            .set('size', pageable.size)
            .set('direction', pageable.direction)
            .set('sort', pageable.sort)
            .set('filterValue', filterValue);
        return this.http.get<Class[]>(environment.endpoint + api.class + api.all + api.student + idStudent, {params});
    }

    getClassByCreator(creator: User) {
        return this.http.get<Class>(environment.endpoint + api.class + api.get + creator);
    }

    getClassById(id: number) {
        return this.http.get<Class>(environment.endpoint + api.class + api.get + id);
    }

    addClass(aClass: Class): Observable<Class> {
        return this.http.post<Class>(environment.endpoint + api.class + api.save, aClass, httpOptions);
    }

    updateClass(aClass: Class): Observable<Class> {
        console.log(JSON.stringify(aClass.id));
        return this.http.put<Class>(environment.endpoint + api.class + api.update + aClass.id, aClass, httpOptions);
    }

    deleteClass(id: number) {
        return this.http.delete<Class>(environment.endpoint + api.class + api.delete + id);
    }

    getClassCount(type: any, id: any) {
        return this.http.get<number>(environment.endpoint + api.class + '/' + type + api.count + id);
    }
}